scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/DatasetController.java

```diff
@@ -0,0 +1,191 @@
+    private final IUserManager userManager;
+
+    @Autowired
+    public DatasetController(DatasetService datasetService,
+                             DataFileService dataFileService,
+                             IDatasetService datasetManager,
+                             IUserManager userManager) {
+        this.datasetService = datasetService;
+        this.dataFileService = dataFileService;
+        this.datasetManager = datasetManager;
+        this.userManager = userManager;
+    }
+
+    @RequestMapping(value = "datasets/{username}", method = RequestMethod.GET)
+    public String get(Model model, @PathVariable String username) {
+        model.addAttribute("dataset", new DatasetForm());
+        List<IDataset> dbDatasets = datasetService.findAllByUsername(username);
+        if (dbDatasets.size() > 0) {
+            model.addAttribute("dbDatasets", dbDatasets);
+        } else {
+            model.addAttribute("noDatasets", "You Have No Datasets Yet");
+        }
+        return "datasets/index";
+    }
+
+    @RequestMapping(value = "datasets/add", method = RequestMethod.POST)
+    public String post(@Valid @ModelAttribute("dataset") DatasetForm datasetForm,
+                       BindingResult result,
+                       Model model,
+                       RedirectAttributes redirectAttributes) {
+        boolean success;
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = authentication.getName();
+        IUser user = userManager.findByUsername(username);
+        datasetForm.setUsername(username);
+
+        if (result.hasErrors()) {
+            // Handle form error - They don't show up automatically with redirects
+            if (datasetForm.getName().isEmpty()) {
+                redirectAttributes.addFlashAttribute("noDatasetName", "Dataset Name Cannot Be Empty");
+            }
+            redirectAttributes.addFlashAttribute("dataset", datasetForm);
+            redirectAttributes.addAttribute("username", username);
+            return "redirect:/datasets/{username}";
+        }
+
+        try {
+            success = datasetService.createDataset(datasetForm.getName(), user);
+        } catch (DatasetException e) {
+            model.addAttribute("dataset", new DatasetForm());
+            redirectAttributes.addFlashAttribute("message", e.getMessage());
+            redirectAttributes.addAttribute("username", username);
+            return "redirect:/datasets/{username}";
+        }
+
+        if (success){
+            redirectAttributes.addFlashAttribute("message", "Dataset successfully created");
+        }
+        return "redirect:/datasets/success";
+    }
+
+    @RequestMapping(value = "datasets/delete", method = RequestMethod.POST)
+    public String delete(RedirectAttributes redirectAttributes, HttpServletRequest request) {
+        Long id = Long.parseLong(request.getParameter("delSetId"));
```

Comment:
this will probably throw an exception if you pass in something else than a number and should be handled.